src/district/ui/component/tx_log.cljs

```diff
@@ -193,11 +193,12 @@
           [:div.transactions
            (dissoc props :tx-cost-currency :transaction-props :transaction-el :no-transactions-props :no-transactions-el)
            (for [{:keys [:hash] :as tx} tx-log-items]
-             ^{:key hash} [transaction-el
-                           (merge
-                            {:tx tx
-                             :tx-cost-currency tx-cost-currency}
-                            transaction-props)])]
+             [transaction-el
+              (merge
+               {:key (or hash (random-uuid)) ;; tx is submitted but not confirmed there is no hash yet
```

Comment:
I don't think it is a good idea to add a random key since it hides the issue from the logs but performance is still the same as no key

Reply:
Sub for this component is coming from :district.ui.web3-tx :txs. Before the on-receipt value look slike this: ```{:created-on #inst "2019-01-22T10:40:15.053-00:00", :transaction-hash "0xf93fdeddc212b5cf5dbe4b14df75566ec6b0198e4ed3af11de937fb4cf9dedfc", :status :tx.status/pending, :tx-log {:name "Mint 1 cards of PepeSmile"}}```

After receiving receipt some extra info is assoc'ed:

```
"0xf93fdeddc212b5cf5dbe4b14df75566ec6b0198e4ed3af11de937fb4cf9dedfc" {24 keys}
{:contract-address nil
    :r "0x24b7a05d822fb4f85d57d4f577585328cdc43ecdd1fdc30bb0fbe060c3fa5e2e"
    :v "0x29"
    :hash "0xf93fdeddc212b5cf5dbe4b14df75566ec6b0198e4ed3af11de937fb4cf9dedfc"
    :transaction-index 36
    :logs [2 items]
    :logs-bloom "0x00 
    :gas-price 1000000000
    :created-on Tue Jan 22 2019 11:40:15 GMT+0100 (Central European Standard Time)
    :block-hash "0xd99fce09ad5426351a37d10f3f9475e1f0e1d78c91dae63331d73ba9386fe38b"
    :value 0
    :gas 6000000
    :tx-log {:name "Mint 1 cards of PepeSmile"}
    :cumulative-gas-used 1954987
     :s "0x6eedc33ea927851e46a6cc2020b4f8feff2c8453c61a484fd9195860319c8716"
     :block-number 4873820
     :tx-costs {2 keys}
     :transaction-hash "0xf93fdeddc212b5cf5dbe4b14df75566ec6b0198e4ed3af11de937fb4cf9dedfc"
     :gas-used 206511
     :status :tx.status/success
     :from "0xc4e0b92df2de77c077d060e49ec63dc196980716"
     :input "0xa0712d680000000000000000000000000000000000000000000000000000000000000001"
     :nonce 215
     :to "0x75940e65ae39aa0491bd31e6c71b3a3e151ed25a"}
```

Point being :hash is not there until transaction receipt is received, but in both cases `:transaction-hash` key is present, so this should fix the warnings.